### PR TITLE
datadog-agent: 6.11.2 -> 7.30.2

### PR DESCRIPTION
**Pull request by john-consumable**
###### Motivation for this change
  #105221 and #101964

Plus, datadog-agent failed to build on master (for me).

###### Things done
datadog-agent is now built as a go module.  However, the build process
is now to be driven by the python `invoke` tool.  While datadog-agent
builds with buildGoModule, some investigation should be done to ensure
we are not missing any integrations or features from the invoke build
system.

This also updates datadog-agent from python 2.7 to 3.x.

One other issue of note: most of the invoke tasks seem to use some git parsing to get version numbers and the like. This package may need to be checked for reproducibility issues now.

- Built on platform(s)
  - [X] x86_64-linux
  - [x] aarch64-linux
  - [X] x86_64-darwin (unsupported platform)
  - [X] aarch64-darwin (unsupported platform)
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [X] (Package updates) Added a release notes entry if the change is major or breaking
  - [X] (Module updates) Added a release notes entry if the change is significant (service changes documented)
  - [X] (Module addition) Added a release notes entry if adding a new NixOS module (no additional NixOS module)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 136109` at 75ab9763 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/db9ab8abd805470936632ec22be157ee">datadog-agent</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by john-consumable** [comment]
Looks like I don't understand the integrations very well. Any help would be appreciated to help CI.

**Comment by john-consumable** [comment]
@ofborg build datadog-agent datadog-process-agent

File: pkgs/tools/networking/dd-agent/integrations-core.nix [hunk 1]
```diff
@@ -41,10 +41,10 @@ let
   src = pkgs.fetchFromGitHub {
     owner = "DataDog";
     repo = "integrations-core";
-    rev = "7e9bebbb5b79ac30c16814ecefdc8f5c63cb4ea4";
-    sha256 = "0yi7dlbd0rkzzl8cag713r86f40vl87aqrj97ral58csnnj7vfzb";
+    rev = "7.30.1";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/dd-agent/integrations-core.nix:44]:
```suggestion
    rev = version;
```

Comment (jsoo1) [reply pkgs/tools/networking/dd-agent/integrations-core.nix]: thank you, done

Comment (john-consumable) [reply pkgs/tools/networking/dd-agent/integrations-core.nix]: Thanks! Done.

**Pull request merged**